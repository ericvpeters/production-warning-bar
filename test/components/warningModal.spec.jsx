import React from 'react'
import { shallow, mount } from 'enzyme';

import WarningModal from '../../src/js/components/warningModal.jsx';
import baseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import injectTapEventPlugin from 'react-tap-event-plugin';
import toJson from 'enzyme-to-json';

describe('WarningModal', () => {
    let component;
    
    beforeAll(() => { injectTapEventPlugin(); });

    beforeEach(() => {
        component = mount( <WarningModal></WarningModal>, {
            context: {
            muiTheme: getMuiTheme(baseTheme)
            }, 
            childContextTypes: {
            muiTheme: React.PropTypes.object.isRequired
            }
        });
    });
    
     
    it("Modal should disappear after clicking on accept button", () => {
        let tree = toJson(component);
        expect(tree).toMatchSnapshot();
        expect(component.find('Dialog').prop('onRequestClose')).toEqual(component.find('Dialog').prop('actions')[0].props['onTouchTap']);
        component.find('Dialog').prop('onRequestClose')();
        expect(component.state('open')).toEqual(false);
        tree = toJson(component);
        expect(tree).toMatchSnapshot();
    });
});
