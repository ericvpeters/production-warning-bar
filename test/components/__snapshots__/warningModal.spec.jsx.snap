exports[`WarningModal Modal should disappear after clicking on accept button 1`] = `
<WarningModal>
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          keyboardFocused={true}
          label="Accept"
          labelPosition="after"
          labelStyle={Object {}}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          primary={true}
          secondary={false} />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    modal={false}
    onRequestClose={[Function]}
    open={true}
    repositionOnUpdate={true}
    style={
      Object {
        "zIndex": 2147483647,
      }
    }
    title="Warning you are in production">
    <RenderToLayer
      open={true}
      render={[Function]}
      useLayerForClickAway={false} />
  </Dialog>
</WarningModal>
`;

exports[`WarningModal Modal should disappear after clicking on accept button 2`] = `
<WarningModal>
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          keyboardFocused={true}
          label="Accept"
          labelPosition="after"
          labelStyle={Object {}}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          primary={true}
          secondary={false} />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    modal={false}
    onRequestClose={[Function]}
    open={false}
    repositionOnUpdate={true}
    style={
      Object {
        "zIndex": 2147483647,
      }
    }
    title="Warning you are in production">
    <RenderToLayer
      open={true}
      render={[Function]}
      useLayerForClickAway={false} />
  </Dialog>
</WarningModal>
`;
